contracts/SignatureVerifirer.sol

```diff
@@ -0,0 +1,30 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol";
```

Comment:
Use the upgradeable variant everywhere:

import "@openzeppelin/contracts-**upgradeable**/utils/cryptography/ECDSA**Upgradeable**.sol";
import "@openzeppelin/contracts-**upgradeable**/utils/cryptography/draft-EIP712**Upgradeable**.sol";